pkg/clusteragent/clusterchecks/dispatcher_configs.go

```diff
@@ -64,7 +64,7 @@ func (d *dispatcher) addConfig(config integration.Config, targetNodeName string)
 	d.store.digestToNode[digest] = targetNodeName
 
 	// Remove config from previous node if found
-	if foundCurrent {
+	if foundCurrent && currentNode != targetNode {
```

Comment:
Any way we can add a comment about the scheduling on the same node ? (the reason why it's important to compare the node and the target)